Yoast/wordpress-seo
# Fixes WooCommerce shop page breadcrumb not being used

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the custom set Breadcrumb title for the WooCommerce Shop page, wouldn't be used.

## Relevant technical choices:

* I had to refactor a bit of the breadcrumbs class to ensure the code didn't get even more complicated to understand.

## Test instructions

This PR can be tested by following these steps:

* Install an old version of the plugin (e.g. 7.6.1).
* Install WooCommerce.
* Ensure you have the Yoast Test Helper installed.
* Ensure you have the breadcrumbs snippet implemented in your theme.

* Set a Shop page in WooCommerce -> Settings -> Products.
* Check out both the Shop archive page and a Product and determine that the default breadcrumb is shown (should be `Products`).

* Go to SEO -> Search Appearance.
* Look for the Products section and set a Breadcrumbs title and save the changes made.
* Check out both the Shop archive page and a Product again and determine that the custom set breadcrumb is shown.

* Switch to this branch. **Do not run the Test Helper just yet!**
* Check out both the Shop archive page and a Product and determine that the default breadcrumb is shown (should be `Products`).
* Go to Tools -> Yoast Test and change the version number for the plugin to something like `7.6.1`.
* Go to the Shop page, switch over to the Advanced meta settings tab and see that the breadcrumb you set in 7.6., is properly migrated.
* Check out both the Shop archive page and a Product again and determine that the custom set breadcrumb is shown.
* Go back to the Shop page, remove the custom set breadcrumb and save the page.
* Check out both the Shop archive page and a Product and determine that the default breadcrumb is shown (should be `Products`).

## Quality assurance


Fixes #10214
